lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -831,26 +830,38 @@ def make_mpiexec_hydra_options(self):
             else:
                 self.log.debug("make_mpiexec_hydra_options: no rmk from HYDRA_RMK %s and hydra_info %s",
                                self.HYDRA_RMK, self.hydra_info)
+
+        default_launcher = getattr(self, 'HYDRA_LAUNCHER', None)
+        if self.options.launcher:
+            launcher = [self.options.launcher]
+            launcher.extend(default_launcher)
         else:
-            launcher = None
-            if getattr(self, 'HYDRA_LAUNCHER', None) is not None:
-                launcher = [x for x in self.HYDRA_LAUNCHER if x in self.hydra_info.get('launcher', [])]
-                if launcher:
-                    self.log.debug("make_mpiexec_hydra_options: HYDRA: launcher %s, using first one", launcher)
-                else:
-                    self.log.debug("make_mpiexec_hydra_options: no launcher from HYDRA_LAUNCHER %s and hydra_info %s",
-                                   self.HYDRA_LAUNCHER, self.hydra_info)
+            launcher = default_launcher
+
+        if launcher:
+            # filter out the launchers that are not in self.hydra_info
+            avail_launcher = [x for x in launcher if x in self.hydra_info.get('launcher', [])]
+            if avail_launcher:
```

Comment:
I would skip this filtering magic, and just straight up either use the value passed to `--launcher` or whatever `HYDRA_LAUNCHER` specifies?

Or, at the very least, don't just silently skip an unknown specified launcher, be very verbose about it (print a warning). And if so, provide an option to force trying to use an unknown launcher.

Long story short, I wouldn't try to be too smart about this, just use what was specified.

Maybe do check what kind of error is printed by Intel MPI when you try to use an unknown launcher; if it's a very obscure error, maybe that's the reason this was added originally.